### ISSUE TITLE
Epoch page: future slots are being refreshed

### ISSUE THREAD

**Issue by sealer3** [otterscan/otterscan#846]
Why do we refresh slots that are still in the future? It seems to send a lot of extra API calls when we are looking at the head epoch. I think we should only try to refresh the slots after the slot time has finally passed.

For example, if we go one or two epochs in the future, we make hundreds of extra API calls for slots with future timestamps, all returning HTTP 404.

**Comment by wmitsuda** [comment]
agree, I think I just couldn't find a good policy for refreshing in SWR, i.e., IIRC it tries to increase time exponentially.

a good policy would be: refresh it after current time > expected time, but I couldn't find a way to program SWR to do it declaratively.

but there is one more gotcha: let's say you refresh it once the expected slot time is reached, but the slot is not there. it could be your browser clock is not behind, so the slot actually didn't come yet; it could be that your CL node didn't receive the slot info from the network, so you need to refresh it again after a certain time.

but it could be that the slot was missed, so once you receive the next slot after that, you may not want to refresh the missed one anymore.

**Comment by wmitsuda** [comment]
I'm also not satisfied with the approach of querying all slots on epoch page, that +the issues in the comment above make me think if it would be better to wait for the Erigon full CL node implementation and add a custom API for batch epoch/slot info and handle everything on node side.

I don't feel like hacking/customizing any of existing CL implementations like I did in Erigon, but if Erigon completes a full CL node implementation that should be ok to do.
